### PR TITLE
Add conda-forge packages to INSTALL

### PR DESCRIPTION
**Pull request by matthiasdiener**
| PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2278](https://github.com/mfem/mfem/pull/2278) | @matthiasdiener | @tzanio | @tzanio + @acfisher + @rw-anderson | 05/27/21 | 06/09/21 | 06/09/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by tzanio** [comment]
Thanks for adding this @matthiasdiener.

Is there an easy way for us to test it?

Tzanio

File: INSTALL [hunk 1]
```diff
@@ -59,6 +59,7 @@ following package managers:
 - Spack, https://github.com/spack/spack
 - OpenHPC, http://openhpc.community
 - Homebrew/Science, https://github.com/Homebrew/homebrew-science (deprecated)
+- Conda-forge, https://conda-forge.org/
```
Comment (tzanio) [inline INSTALL:62]:
```suggestion
- Conda-forge, https://conda-forge.org
- Homebrew/Science, https://github.com/Homebrew/homebrew-science (deprecated)
```

**Comment by matthiasdiener** [comment]
> Thanks for adding this @matthiasdiener.
> 
> Is there an easy way for us to test it?
> 
> Tzanio

I'm not sure how familiar you are with conda and conda-forge, but a simple installation could look like this:

```bash
$ wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
$ bash Miniforge3-Linux-x86_64.sh
$ source $HOME/miniforge3/activate
$ conda install mfem
```

The way the mfem package is built can be found here: https://github.com/conda-forge/mfem-feedstock/ .

In particular, https://github.com/conda-forge/mfem-feedstock/blob/master/recipe/build.sh is the build script for mfem. 

Please let me know if you have any suggestions for the build (I am the maintainer of https://github.com/conda-forge/mfem-feedstock/).

**Comment by tzanio** [comment]
@rw-anderson or @sshiraiwa, are you interested in giving the conda package a try?

**Comment by sshiraiwa** [comment]
Yes, I will...over the weekend.

**Comment by tzanio** [comment]
thanks @sshiraiwa !

**Comment by sshiraiwa** [comment]
This seems work fine.

> ```shell
> $ wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
> $ bash Miniforge3-Linux-x86_64.sh
> $ source $HOME/miniforge3/activate
> $ conda install mfem
> ```
For me, it was 
```
source $HOME/miniforge3/bin/activate
```

Extra note on Python wrapper. This package installs mfem with mpi/hypre/metis on, and the binary package in PyPI (which is build as serial) is not suitable to use.  One can still build PyMFEM from source using the following step (tested on linux)

```
conda install openmpi-mpicc
conda install openmpi-mpicxx
pip3 install mpi4py
pip3 install numpy
pip3 install scipy
pip3 download mfem --no-binary mfem
tar -zxvf mfem-4.2.0.16.tar.gz
cd mfem-4.2.0.16
python setup.py install --mfem-prefix=$CONDA_PREFIX  --hypre-prefix=$CONDA_PREFIX --metis-prefix=$CONDA_PREFIX  --no-serial --mfem-prefix-no-swig --with-parallel
```
After this, I was able to run ex1p.py in parallel.
```
cp -r /home/shiraiwa/src/mfem/data ./   ### need to copy data folder for mesh from mfem
cd examples
mpirun -np 4 python ex1p.py
```

**Review by sshiraiwa — APPROVED** [review]

**Review by sshiraiwa — COMMENTED** [review]
It may be helpful to say "(Pre-build binary linked with OpenMPI, Hypre and Metis)"?

**Comment by matthiasdiener** [comment]
> This seems work fine.
> 
> > ```shell
> > $ wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
> > $ bash Miniforge3-Linux-x86_64.sh
> > $ source $HOME/miniforge3/activate
> > $ conda install mfem
> > ```
> 
> For me, it was
> 
> ```
> source $HOME/miniforge3/bin/activate
> ```
> 
> Extra note on Python wrapper. This package installs mfem with mpi/hypre/metis on, and the binary package in PyPI (which is build as serial) is not suitable to use. One can still build PyMFEM from source using the following step (tested on linux)
> 
> ```
> conda install openmpi-mpicc
> conda install openmpi-mpicxx
> pip3 install mpi4py
> pip3 install numpy
> pip3 install scipy
> pip3 download mfem --no-binary mfem
> tar -zxvf mfem-4.2.0.16.tar.gz
> cd mfem-4.2.0.16
> python setup.py install --mfem-prefix=$CONDA_PREFIX  --hypre-prefix=$CONDA_PREFIX --metis-prefix=$CONDA_PREFIX  --no-serial --mfem-prefix-no-swig --with-parallel
> ```
> 
> After this, I was able to run ex1p.py in parallel.
> 
> ```
> cp -r /home/shiraiwa/src/mfem/data ./   ### need to copy data folder for mesh from mfem
> cd examples
> mpirun -np 4 python ex1p.py
> ```

Thanks @sshiraiwa! I will add the Python wrapper to a future release of the mfem conda package.

**Comment by matthiasdiener** [comment]
> It may be helpful to say "(Pre-build binary linked with OpenMPI, Hypre and Metis)"?

This might be helpful, but please mention that the mfem package is built for OpenMPI as well as mpich.

File: INSTALL [hunk 2]
```diff
@@ -58,7 +58,7 @@ following package managers:
 
 - Spack, https://github.com/spack/spack
 - OpenHPC, http://openhpc.community
-- Conda-forge, https://conda-forge.org
+- Conda-forge, https://conda-forge.org (pre-built binaries linked with OpenMPI/Mpich, Hypre, and METIS)
```
Comment (tzanio) [inline INSTALL:61]:
```suggestion
- Conda-forge, https://conda-forge.org (pre-built binaries linked with OpenMPI/MPICH, hypre, and METIS)
```

**Review by acfisher — APPROVED** [review]

**Comment by rw-anderson** [comment]
Seemed to work fine on my ubuntu 18.04 box.  Thanks!

**Review by rw-anderson — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Review by tzanio — APPROVED** [review]
Thanks so much for this contribution @matthiasdiener!

We can mention this as a news item at mfem.org if you want. Feel free to open a PR at https://github.com/mfem/web if you are interested.

**Pull request merged**